demos/speech_recognition_demo/python/speech_recognition_demo.py

```diff
@@ -25,28 +24,35 @@ def build_argparser():
     parser.add_argument('-d', '--device', default='CPU', type=str,
                         help="Optional. Specify the target device to infer on, for example: CPU, GPU, FPGA, HDDL, MYRIAD or HETERO. "
                              "The sample will look for a suitable IE plugin for this device. (default is CPU)")
-    parser.add_argument('-m', '--model', type=str, metavar="FILENAME", required=True,
-                        help="Path to an .xml file with a trained model (required)")
-    parser.add_argument('-L', '--lm', type=str, metavar="FILENAME",
-                        help="path to language model file (optional)")
-    parser.add_argument('-p', '--profile', type=str, metavar="NAME", required=True,
-                        help="Choose pre/post-processing profile: "
-                             "mds06x_en for Mozilla DeepSpeech v0.6.x, "
-                             "mds07x_en or mds08x_en for Mozilla DeepSpeech v0.7.x/x0.8.x, "
-                             "other: filename of a YAML file (required)")
-    parser.add_argument('-b', '--beam-width', type=int, default=500, metavar="N",
-                        help="Beam width for beam search in CTC decoder (default 500)")
-    parser.add_argument('-c', '--max-candidates', type=int, default=1, metavar="N",
-                        help="Show top N (or less) candidates (default 1)")
-
-    parser.add_argument('-l', '--cpu_extension', type=str, metavar="FILENAME",
-                        help="Optional. Required for CPU custom layers. "
-                             "MKLDNN (CPU)-targeted custom layers. Absolute path to a shared library with the"
-                             " kernels implementations.")
+
+    architecture_type_subparsers = parser.add_subparsers(dest='architecture_type', required=True,
```

Comment:
You've probably seen me say this in the MelGAN demo PR, but I don't think subcommands are the best way to implement this. Especially since we already do have a pattern for selecting the architecture type - an `-architecture_type` option.

Reply:
`-architecture_type` is redundant there because it would follow `bool(args.model_decoder)`. Updated